beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -0,0 +1,305 @@
+//! This module implements an optimisation to fetch blobs via JSON-RPC from the EL.
+//! If a blob has already been seen in the public mempool, then it is often unnecessary to wait for
+//! it to arrive on P2P gossip. This PR uses a new JSON-RPC method (`engine_getBlobsV1`) which
+//! allows the CL to load the blobs quickly from the EL's blob pool.
+//!
+//! Once the node fetches the blobs from EL, it then publishes the remaining blobs that hasn't seen
+//! on P2P gossip to the network. From PeerDAS onwards, together with the increase in blob count,
+//! broadcasting blobs requires a much higher bandwidth, and is only done by high capacity
+//! supernodes.
+use crate::kzg_utils::blobs_to_data_column_sidecars;
+use crate::observed_data_sidecars::ObservableDataSidecar;
+use crate::{metrics, AvailabilityProcessingStatus, BeaconChain, BeaconChainTypes, BlockError};
+use execution_layer::json_structures::BlobAndProofV1;
+use execution_layer::Error as ExecutionLayerError;
+use itertools::Either;
+use lighthouse_metrics::{inc_counter, TryExt};
+use rand::prelude::SliceRandom;
+use slog::{debug, error};
+use ssz_types::FixedVector;
+use state_processing::per_block_processing::deneb::kzg_commitment_to_versioned_hash;
+use std::sync::Arc;
+use std::time::Duration;
+use types::blob_sidecar::{BlobSidecarError, FixedBlobSidecarList};
+use types::{
+    BeaconStateError, BlobSidecar, DataColumnSidecarList, EthSpec, FullPayload, Hash256,
+    SignedBeaconBlock, SignedBeaconBlockHeader,
+};
+
+/// Number of batches that supernodes split data columns into during publishing.
+pub const SUPERNODE_DATA_COLUMN_PUBLICATION_BATCHES: usize = 4;
+
+/// The delay applied by supernodes between the sending of each data column batch.
+pub const SUPERNODE_DATA_COLUMN_PUBLICATION_BATCH_INTERVAL: Duration = Duration::from_millis(200);
```

Comment:
These (batch count + interval) are key parameters that we are just yolo guessing, should this be a flag?

Reply:
yeh

Reply:
Addressed in 0a9d5a0f61b4114dff93fe88c46ca4b97dde75f2